docs/docker-actions.md

```diff
@@ -0,0 +1,154 @@
+
+  ```
+  cat dockerSkeleton/example.c
+  ```
+  ```c
+  #include <stdio.h>
+  int main(int argc, char *argv[]) {
+      printf("This is an example log message from an arbitrary C program!\n");
+      printf("{ \"msg\": \"Hello from arbitrary C program!\", \"args\": %s }",
+             (argc == 1) ? "undefined" : argv[1]);
+  }
+  ```
+
+  You can modify this file as needed, or, add additional code and dependencies to the Docker image.
+  In case of the latter, you may need to tweak the `Dockerfile` as necessary to build your executable.
+  The binary must be located inside the container at `/action/exec`.
+
+  The executable receives a single argument from the command line. It is a string serialization of the JSON
+  object representing the arguments to the action. The program may log to `stdout` or `stderr`.
+  By convention, the last line of output _must_ be a stringified JSON object which represents the result of the action.
+
+3. Build the Docker image and upload it using a supplied script. You must first run `docker login` to authenticate, and then run the script with a chosen image name.
+
+  ```
+  docker login -u janesmith -p janes_password
+  ```
+  ```
+  cd dockerSkeleton
+  ```
+  ```
+  ./buildAndPush.sh janesmith/blackboxdemo
+  ```
+
+  Notice that part of the example.c file is compiled as part of the Docker image build process, so you do not need C compiled on your machine.
+  In fact, unless you are compiling the binary on a compatible host machine, it may not run inside the container since formats will not match.
+
+  Your Docker container may now be used as an OpenWhisk action.
+
+
+  ```
+  wsk action create example --docker janesmith/blackboxdemo
+  ```
+
+  Notice the use of `--docker` when creating an action. Currently all Docker images are assumed to be hosted on Docker Hub.
+  The action may be invoked as any other OpenWhisk action.
+
+  ```
+  wsk action invoke --result example --param payload Rey
+  ```
+  ```json
+  {
+      "args": {
+          "payload": "Rey"
+      },
+      "msg": "Hello from arbitrary C program!"
+  }
+  ```
+
+  To update the Docker action, run buildAndPush.sh to upload the latest image to Docker Hub. This will allow the system to pull your new Docker image the next time it runs the code for your action.
+  If there are no warm containers any new invocations will use the new Docker image.
+  However, if there is a warm container using a previous version of your Docker image, any new invocations will continue to use that image unless you run `wsk action update`. This will indicate to the system that for new invocations it should execute a docker pull to get your new Docker image.
+
+  ```
+  ./buildAndPush.sh janesmith/blackboxdemo
+  ```
+  ```
+  wsk action update example --docker janesmith/blackboxdemo
+  ```
+
+  You can find more information about creating Docker actions in the [References](./reference.md#docker-actions) section.
```

Comment:
The "Docker actions" section is no longer in reference.md, but I can't find it in docker-actions.md either!